### PR TITLE
chore: add moore voting algo

### PR DESCRIPTION
**Pull request by abhipadhy**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+//majorty element of an array
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Arrays/boyer_more.cpp:1]: correct the spelling of majority

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
+//majorty element of an array 
+// element with frequency > (n/2) {n: size of array}
+//time complexity : O(n) || space complexity : O(1)
+
+
+#include<iostream>
+#include<stdio.h>
+
+
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Arrays/boyer_more.cpp:9]: please remove the too much extra space

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
+//majorty element of an array 
+// element with frequency > (n/2) {n: size of array}
+//time complexity : O(n) || space complexity : O(1)
+
+
+#include<iostream>
+#include<stdio.h>
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Arrays/boyer_more.cpp]: don't use stdio.h here

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    else{
+        cout<<"no majority element found\n";
+    }
+}
+
+
+int majelement(int arr[],int n){ // maj element function
+
+     int len = n;
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Arrays/boyer_more.cpp:36]: no need to assign n in len well n is local variable

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+#include<stdio.h>
+
+
+using namespace std;
+
+
+void ismaj(int arr[],int n,int exp_maj){
+
+    int i=0;
+    int len = n;
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Arrays/boyer_more.cpp:16]: please add the function description and some comment so it look good

Comment (abhipadhy) [reply algorithms/CPlusPlus/Arrays/boyer_more.cpp:16]: yes updating

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    if(count>(n/2)){
+        cout<<"majority element : "<<arr[exp_maj];
+    }
+    else{
+        cout<<"no majority element found\n";
+    }
+}
+
+
+int majelement(int arr[],int n){ // maj element function
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Arrays/boyer_more.cpp]: add the function description i.e.
> // maj element function
 // function to find  major element in an array

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+            count =1; // resetting count
+         }
+    }
+
+    return ans_index;// expected majority element
+} 
+
+
+int main(){ // main function begins here
+    int arr[5] = {2,2,2,5,6};
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Arrays/boyer_more.cpp:52]: better to take input from user

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp (hunk 6 above)
Comment (abhipadhy) [reply algorithms/CPlusPlus/Arrays/boyer_more.cpp]: updated

**Review by UG-SEP — CHANGES_REQUESTED** [review]
fell free to ask any doubt

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp (hunk 7 above)
Comment (abhipadhy) [reply algorithms/CPlusPlus/Arrays/boyer_more.cpp:52]: is the code fine now?

**Comment by UG-SEP** [comment]
hey @ming-tsai  
@abhipadhy has added java code too so  i  check the cpp file 
but assign one more reviewer so he/she can check the java one

File: algorithms/CPlusPlus/Arrays/boyer_more.cpp (hunk 7 above)
Comment (abhipadhy) [reply algorithms/CPlusPlus/Arrays/boyer_more.cpp]: hey @UG-SEP i have updated the user input part too kindly review the code

**Comment by UG-SEP** [comment]
@abhipadhy please update the readme.md 
hint: add the program name and link in array block

**Comment by abhipadhy** [comment]
> @abhipadhy please update the readme.md
> hint: add the program name and link in array block
> [click me to update readme](https://github.com/MakeContributions/DSA/blob/main/algorithms/CPlusPlus/README.md)

@UG-SEP i have made all necessary changes according to you kindly merge my pr

**Comment by UG-SEP** [comment]
> > @abhipadhy please update the readme.md
> > hint: add the program name and link in array block
> > [click me to update readme](https://github.com/MakeContributions/DSA/blob/main/algorithms/CPlusPlus/README.md)
> 
> @UG-SEP i have made all necessary changes according to you kindly merge my pr

c+
Please add your program name and it's link in readme link given above

**Comment by abhipadhy** [comment]
> > > @abhipadhy please update the readme.md
> > > hint: add the program name and link in array block
> > > [click me to update readme](https://github.com/MakeContributions/DSA/blob/main/algorithms/CPlusPlus/README.md)
> > 
> > 
> > @UG-SEP i have made all necessary changes according to you kindly merge my pr
> 
> c+
> Please add your program name and it's link in readme link given above

@UG-SEP updated readme

**Comment by ming-tsai** [comment]
Hi @abhipadhy,
Please fix the linter errors, and delete the file `algorithms/CPlusPlus/Arrays/boyer_more.exe`, the binary file is not allower

**Comment by ming-tsai** [comment]
```
Error:    3:26  error  Trailing spaces not allowed                                                                    no-trailing-spaces
Error:    5:1   error  Trailing spaces not allowed                                                                    no-trailing-spaces
Error:    7:1   error  Opening curly brace does not appear on the same line as controlling statement                  brace-style
Error:    8:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:    8:5   error  Split 'let' declarations into multiple statements                                              one-var
Error:    8:9   error  Identifier 'maj_index' is not in camel case                                                    camelcase
Error:    9:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   10:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   11:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   11:9   error  Expected { after 'if' condition                                                                curly
Error:   12:1   error  Expected indentation of 6 spaces but found 12                                                  indent
Error:   13:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   13:9   error  Expected { after 'else'                                                                        curly
Error:   14:1   error  Expected indentation of 6 spaces but found 12                                                  indent
Error:   16:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   17:1   error  Expected indentation of 6 spaces but found 12                                                  indent
Error:   17:13  error  Identifier 'maj_index' is not in camel case                                                    camelcase
Error:   18:1   error  Expected indentation of 6 spaces but found 12                                                  indent
Error:   19:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   20:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   21:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   25:1   error  This line has a length of 99. Maximum allowed is 80                                            max-len
Error:   25:10  error  'isMaj' is defined but never used                                                              no-unused-vars
Error:   26:1   error  Opening curly brace does not appear on the same line as controlling statement                  brace-style
Error:   27:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   27:5   error  Split 'let' declarations into multiple statements                                              one-var
Error:   28:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   29:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   29:9   error  Expected { after 'if' condition                                                                curly
Error:   30:1   error  Expected indentation of 6 spaces but found 12                                                  indent
Error:   31:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   32:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   32:5   error  Expected { after 'if' condition                                                                curly
Error:   33:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   34:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   34:5   error  Expected { after 'else'                                                                        curly
Error:   35:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   39:1   error  Opening curly brace does not appear on the same line as controlling statement                  brace-style
Error:   39:1   error  Block must not be padded by blank lines                                                        padded-blocks
Error:   40:1   error  Trailing spaces not allowed                                                                    no-trailing-spaces
Error:   41:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   41:5   error  Unexpected var, use let or const instead                                                       no-var
Error:   43:1   error  Trailing spaces not allowed                                                                    no-trailing-spaces
Error:   44:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   44:5   error  Expected { after 'if' condition                                                                curly
Error:   45:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   45:21  error  Strings must use singlequote                                                                   quotes
Error:   45:34  error  Strings must use singlequote                                                                   quotes
Error:   46:1   error  Expected indentation of 2 spaces but found 4                                                   indent
Error:   46:5   error  Expected { after 'else'                                                                        curly
Error:   47:1   error  Expected indentation of 4 spaces but found 8                                                   indent
Error:   47:21  error  Strings must use singlequote                                                                   quotes
Error:   50:1   error  Trailing spaces not allowed                                                                    no-trailing-spaces
Error:   51:1   error  Unexpected var, use let or const instead                                                       no-var
Error:   51:5   error  'arr' is assigned a value but never used                                                       no-unused-vars
Error:   51:12  error  There should be no space after '['                                                             array-bracket-spacing
Error:   51:20  error  A space is required after ','                                                                  comma-spacing
Error:   51:28  error  There should be no space before ']'                                                            array-bracket-spacing
Error:   52:1   error  Unexpected var, use let or const instead                                                       no-var
Error:   54:1   error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
```

**Comment by abhipadhy** [comment]
@ming-tsai  i have removed the majorityelement.js file , i"ll will push the code for that afterwards .
for the time being i am just going for the cpp part for the PR

**Comment by UG-SEP** [comment]
image
please correct the number sequence there are two 3

**Review by UG-SEP — CHANGES_REQUESTED** [review]
please change the request else everything is fine

**Comment by abhipadhy** [comment]
> please change the request else everything is fine

done . kindly merge my PR

**Review by UG-SEP — APPROVED** [review]
@abhipadhy Thank for contributing 🙏
Happy to be a contributor like you 😀😀
Hope you contribute more to this project 🤞
Thanks 😊😊😊

**Comment by UG-SEP** [comment]
@ming-tsai please check it

**Pull request merged**